e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js

```diff
@@ -1,38 +1,75 @@
-import { PENDING_RECONCILIATION, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { FEE_RECORD_STATUS, FeeRecordEntityMockBuilder, PENDING_RECONCILIATION, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
 import pages from '../../../../pages';
 import USERS from '../../../../../fixtures/users';
 import { NODE_TASKS } from '../../../../../../../e2e-fixtures';
+import relative from '../../../../relativeURL';
+import { errorSummary } from '../../../../partials';
+import { getMatchingTfmFacilitiesForFeeRecords } from '../../../../../support/utils/getMatchingTfmFacilitiesForFeeRecords';
 
 context('When fee record correction feature flag is enabled', () => {
-  context('PDC_RECONCILE users can create record correction requests', () => {
-    const bankId = '961';
-    const reportId = 1;
+  const bankId = '961';
+  const reportId = 1;
 
-    const report = UtilisationReportEntityMockBuilder.forStatus(PENDING_RECONCILIATION).withId(reportId).withBankId(bankId).build();
+  const report = UtilisationReportEntityMockBuilder.forStatus(PENDING_RECONCILIATION).withId(reportId).withBankId(bankId).build();
+  const feeRecordAtToDoStatus = FeeRecordEntityMockBuilder.forReport(report).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO).build();
 
-    const { premiumPaymentsTab } = pages.utilisationReportPage;
+  const { premiumPaymentsTab } = pages.utilisationReportPage;
 
-    before(() => {
-      cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
-      cy.task(NODE_TASKS.REINSERT_ZERO_THRESHOLD_PAYMENT_MATCHING_TOLERANCES);
-    });
+  before(() => {
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+    cy.task(NODE_TASKS.REINSERT_ZERO_THRESHOLD_PAYMENT_MATCHING_TOLERANCES);
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecordAtToDoStatus]);
 
+    const matchingTfmFacilities = getMatchingTfmFacilitiesForFeeRecords([feeRecordAtToDoStatus]);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, matchingTfmFacilities);
+  });
+
+  after(() => {
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+    cy.task(NODE_TASKS.REINSERT_ZERO_THRESHOLD_PAYMENT_MATCHING_TOLERANCES);
+  });
+
+  context('PDC_READ users cannot create record correction requests', () => {
     beforeEach(() => {
-      cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+      pages.landingPage.visit();
+      cy.login(USERS.PDC_READ);
+
+      cy.visit(`utilisation-reports/${reportId}`);
+    });
 
+    it('should not be able to initiate record correction request', () => {
+      premiumPaymentsTab.createRecordCorrectionRequestButton().should('not.exist');
+    });
+  });
```

Comment:
Can/should we add a test for a PDC_READ user manually navigating to the page (so bypassing the submit button)?

Reply:
It's quite simple to do so so have added it 😄